src/hooks/useBankTransactions/useBankTransactions.tsx

```diff
@@ -304,16 +324,13 @@ export const useBankTransactions: UseBankTransactions = params => {
   }
 
   const removeAfterCategorize = (bankTransaction: BankTransaction) => {
-    /*
-    Removed 2024-08-12 by @doneel as part of unifying transaction lists in LAY-298
     const updatedData = rawResponseData?.map(page => {
```

Comment:
This block is still going to cause us problems: we only want to do the client-side removal animation from the list a filter no longer applies. The only situation I can think of is if 
1. You have a filter set to uncategorized only.
2. You categorize a transaction.

None of the other data on a transaction that can be used by filters (`amount`, `account`, `direction`, `dateRange`) can be affected by user action.


So I think we'll have to do a special one-off case here where we only want to apply this affect in that situation. In my ideal world we have generalized method of "did filters previously _allow_ this transaction but now should _disallow_", but that would require
1. Keeping our client-side filter logic
2. Predicting the state of a transaction before we get a response back from the server.

This is to say, I don't think we're going to get a super generalizable way of doing this, so we should basically hardcode 
```ts
    if (
      filters?.categorizationStatus === DisplayState.review && 
      ReviewCategories.includes(bankTransaction.categorization_status)
    ) {
      //do the removal
    }
```

Reply:
Turns out I needed to do this multiple places so I added it as shouldHideAfterCategorize in useBankTransactions. I used it and now we should only do the client-side removal animation when appropriate.